### DISCUSSION TITLE
multi local secrets

### DISCUSSION THREAD

**Discussion by misch42** [strongswan/strongswan#1735] [General]
Hi,

I want to set up different tunnels from one host to different partners. With every partner we want to identify with our public IP address. With every partner we want to have a different secret.

How to achieve different secrets for the same local id?

I read the doc and set up
secrets.ike-A.id = "local ip"
secrets.ike-A.secret = "secret with A"
secrets.ike-B.id = "local ip"
secrets.ike-B.secret = "secret with B"

is the following syntax correct?

conn.A.local-A.id = "local ip"
conn.A.local-B.id = "local ip"

Michael.

**Comment by tobiasbrunner** [comment]
> How to achieve different secrets for the same local id?

Don't configure the local IP/identity but the remote IP/identity instead. That is, use:

secrets.ike-A.id = "remote ipA"
secrets.ike-B.id = "remote ipB"



> is the following syntax correct?
> 
> conn.A.local-A.id = "local ip"
> conn.A.local-B.id = "local ip"

Hm, no, that doesn't seem right. With that syntax you'd configure multiple local authentication rounds.

You probably want something like:

conn.A.local.id = "local ip"
conn.B.local.id = "local ip"

Or just a single connection will work fine, too, if the rest of the config is the same anyway.

**Comment by misch42** [comment]
Of course we can configure different identities and different secrets for the remote ends. They are different IP addresses, so this is easy. This part is simple.

The problem is different secrets for for the local id, that is the local IP address. We have two customers and need two different VPN tunnels. we want to identify with our IP address but have different secrets for both VPN tunnels.

**Reply by tobiasbrunner** [reply]
Your local identity doesn't matter as long as you associate the secrets only with the remote identity.
